planner/cascades/transformation_rules.go

```diff
@@ -1286,3 +1288,38 @@ func (r *MergeAdjacentSelection) OnTransform(old *memo.ExprIter) (newExprs []*me
 	newSelExpr.SetChildren(childGroups...)
 	return []*memo.GroupExpr{newSelExpr}, true, false, nil
 }
+
+// MergeAdjacentLimit merge the adjacent limit.
+type MergeAdjacentLimit struct {
+	baseRule
+}
+
+// NewRuleMergeAdjacentLimit creates a new Transformation MergeAdjacentLimit.
+// The pattern of this rule is `Limit->Limit->X`.
+func NewRuleMergeAdjacentLimit() Transformation {
+	rule := &MergeAdjacentLimit{}
+	rule.pattern = memo.BuildPattern(
+		memo.OperandLimit,
+		memo.EngineAll,
+		memo.NewPattern(memo.OperandLimit, memo.EngineAll),
+	)
+	return rule
+}
+
+// OnTransform implements Transformation interface.
+// This rule tries to merge adjacent limit..
+func (r *MergeAdjacentLimit) OnTransform(old *memo.ExprIter) (newExprs []*memo.GroupExpr, eraseOld bool, eraseAll bool, err error) {
```

Comment:
we need a `Match()` function to ensure that the two `Limit` operators are both executed in the same execution engine. For example:

* case 1(positive): both `Limit`s in `EngineTiDB` can be merged to one `Limit`
* case 2(negative): one `Limit` in the `EngineTiDB`, the other `Limit` in the `EngineTiKV`, these two `Limit` can not merged.

Reply:
Case2 won't happen in the current framework. There will be a `Gather` between `EngineTiDB` and `EngineTiKV`. So we will never meet the pattern `TiDBLimit -> TiKVLimit`. 

@SeaRise, sorry for the inconvenience, could you please revert this change? We don't need this `Match` here, actually.

Reply:
ok